network/src/network.rs

```diff
@@ -468,31 +469,46 @@ impl NetworkState {
     /// this method is intent to check observed addr by dial to self
     pub(crate) fn try_dial_observed_addrs(&self, p2p_control: &ServiceControl) {
         let mut pending_observed_addrs = self.pending_observed_addrs.write();
-        let public_addrs = { self.public_addrs.read().clone() };
-        for addr in pending_observed_addrs
-            .drain()
-            .chain(public_addrs.into_iter())
-        {
+        if pending_observed_addrs.is_empty() {
+            let addrs = self.public_addrs.write();
+            let addr = addrs.iter().next().unwrap();
             trace!("try dial observed addr: {:?}", addr);
-            if let Err(err) = self.dial_inner(
+            match self.dial_inner(
                 p2p_control,
                 self.local_peer_id(),
-                addr,
+                addr.clone(),
                 TargetProtocol::Single(SupportProtocols::Identify.protocol_id()),
                 true,
             ) {
-                debug!("try_dial_observed_addrs error {}", err);
+                Err(err) => {
+                    debug!("try_dial_observed_addrs error {}", err);
```

Comment:
Just a thinly veiled criticism, the key word `error` in `debug` logs is terrible.
- It makes `error` to be worthless.
  When I enable `log=trace` to debug, then `grep -i error` is meaningless.
  I have to use a complex filter on a huge file, and wait more time.

- Cause panic to users who enable `log=debug`.
  Not all users could distinguish which `error` are insignificant and which are urgent. 
  Yes, I can filter the log records by log level, but I have to use a more complex logs filter, which is terrible with a huge file.